src/librustc/ty/context.rs

```diff
@@ -1581,6 +1581,32 @@ impl<'tcx> TyCtxt<'tcx> {
     pub fn has_strict_asm_symbol_naming(&self) -> bool {
         self.gcx.sess.target.target.arch.contains("nvptx")
     }
+
+    /// Determine if this function gets a shim to catch panics and abort.
+    pub fn abort_on_panic_shim(&self, fn_def_id: DefId, abi: abi::Abi) -> bool {
+        use abi::Abi;
+
+        // Not callable from C, so we can safely unwind through these
+        if abi == Abi::Rust || abi == Abi::RustCall { return false; }
```

Comment:
This line is the cause for https://github.com/rust-lang/rust/issues/63883. Should I just remove it?

Reply:
For the `None` case below, we need a default behavior, which right now depends on the Abi string, so I'd rather move this down there for now.

Reply:
All right, I moved this check down to the default branch.